### ISSUE TITLE
[Request] add :NOTER_DOCUMENT: to all notes/annotation

### ISSUE THREAD

**Issue by FirasSadiyah** [weirdNox/org-noter#37]
Hi, 

Is it possible to include the `:NOTER_DOCUMENT:` field in addition to the `:NOTER_PAGE:` for all the notes/annotations in the `:PROPERTIES:` drawer. 

The idea is to have independent pieces of information (LEGO blocks) which are not restricted to the header of the original paper but can be moved/copied to a new tree in org-mode (e.g. when writing a paper) and keep a reference (and possibly a link) to the original paper.

Many thanks, 

Firas

**Comment by weirdNox** [comment]
Hm, I understand what you want but, right now, that is not directly doable, because you may have nested documents and then org-noter would create a session on the note itself instead of on the root heading.

Does it need to be the property NOTER_DOCUMENT specifically?

**Comment by FirasSadiyah** [comment]
I need to do more testing, but when I copied the `NOTER_DOCUMENT` from the root header (the paper) into the notes and then moved the notes to a different root (within the same document), I was able to reserve the connection with the cited page, which is great. 

It doesn't have to be the property `NOTER_DOCUMENT` as long as org-noter is able to link the page of the cited pdf using that property.  

This is mainly for when one finishes summarising the paper and then starts to produce their own piece of writing. The shuffling of these little notes beyond the header tree is truly beneficial for the flow of thoughts during the writing process while keeping the necessary citation.

**Comment by weirdNox** [comment]
I believe this implements what you want, after setting `org-noter-doc-property-in-notes` to `t`. Please try it, and report back if it breaks features (I have solved some problems this introduced, but I might have not caught them all!) 

Anyway, notice that when moving the notes out, they will act like the root of the notes, which means they will stop being a "note".

**Comment by FirasSadiyah** [comment]
Many thanks for implementing this one. I do appreciate it. 

It is working as it supposed to be. However, I noticed that when I take the note out and then scroll through the linked pdf, the `NOTER_PAGE` property is changed to reflect the current position in the PDF. 

Is changing the root's `NOTER_PAGE` required for the auto-save feature? Setting `(setq org-noter-auto-save-last-location nil)` didn't help. 

Is there a way to stop changing the `NOTER_PAGE` in this case?

**Comment by weirdNox** [comment]
I believe this fixes both your problem and another I just found when using the command `org-noter-sync-current-note`

**Comment by FirasSadiyah** [comment]
Many thanks for your help. 
I am afraid that I am not able to get this one working. 

So, I set `org-noter-auto-save-last-location` to `t` and then to `nil`. In both cases, the `:NOTER_PAGE:` changes from the precise location to an integer representing the page number as soon as I issue `M-x: org-noter`. I can see the `:NOTER_AUTO_SAVE_LAST_LOCATION: nil` on every note that I created, but that is not preventing the page number from changing when I scroll through the PDF. 

Any idea what I might be doing wrong?
